dnsmenace.py

```
@@ -195,6 +261,157 @@ async def query_dns_simple(
+
+async def get_ip_geolocation(ip: str) -> dict | None:
+    """Get geolocation info for an IP address."""
+    async with httpx.AsyncClient(timeout=10.0) as client:
+        try:
+            response = await client.get(f"http://ip-api.com/json/{ip}")
+            if response.status_code == 200:
+                data = response.json()
+                if data.get("status") == "success":
+                    return data
+        except Exception:
+            pass
```

Ruff (0.14.14)

[error] `try`-`except`-`pass` detected, consider logging the exception

(S110)